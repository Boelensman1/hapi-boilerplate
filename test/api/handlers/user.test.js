const path = require('path')
const request = require('supertest')
const setUpHandlerTest = require('test/api/setUpHandlerTest')

const seedLocation = path.join(__dirname, 'user.seed.yaml')

describe('Test /user route', () => {
  test('Get users', async () => {
    const { listener } = await setUpHandlerTest(seedLocation)

    const response = await request(listener)
      .get('/users')
      .expect('Content-Type', /json/)
      .expect(200)

    // should return an array
    response.body.forEach((user) => {
      expect(user).toMatchSnapshot({ createdAt: expect.any(String) })
    })
  })

  test('Delete a user', async () => {
    const { listener, models } = await setUpHandlerTest(seedLocation)

    const userCount = (await models.user.query()).length

    const response = await request(listener)
      .delete('/user/10')
      .expect('Content-Type', /json/)
      .expect(200)

    // should return an array
    expect(response.body).toEqual({ result: 'OK' })

    // there should be one less user
    const users = await models.user.query()
    expect(users.length).toBe(userCount - 1)
  })
})
