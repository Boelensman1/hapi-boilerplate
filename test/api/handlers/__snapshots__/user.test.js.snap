// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test /user route Get users 1`] = `
Array [
  Object {
    "createdAt": "1970-01-01T00:00:00.000Z",
    "id": 0,
    "loggedInAt": null,
    "roleId": 0,
    "updatedAt": null,
    "username": "SYSTEM",
  },
  Object {
    "createdAt": "2029-12-31T22:00:00.000Z",
    "id": 10,
    "loggedInAt": null,
    "roleId": 200,
    "updatedAt": null,
    "username": "testUser1",
  },
  Object {
    "createdAt": "2029-12-31T22:00:00.000Z",
    "id": 11,
    "loggedInAt": null,
    "roleId": 200,
    "updatedAt": null,
    "username": "testUser2",
  },
]
`;
